### PR TITLE
Dtsbps 463 repo service

### PR DESCRIPTION
**Pull request by gauravchawla73463**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSBPS-463

### Change description ###
parameter dan added for search for /zip-files endpoint

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 1]
```diff
@@ -42,6 +53,30 @@ public ZipFileStatus getStatusFor(String zipFileName) {
         );
     }
 
+    public List<ZipFileStatus> getStatusByDcn(String documentControlNumber) throws InvalidParameterException {
+
+        if (documentControlNumber.length() < 6) {
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:58]: 6 should be a constant

Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java]: made it constant

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ScannableItemRepository.java [hunk 2]
```diff
@@ -1,9 +1,28 @@
...
 
+    /**
+     * Finds envelope with a blob not deleted for a given container and zip file name.
+     *
+     * @param documentControlNumber of scannable item.
+     * @return A list of envelopes.
+     */
+    @Query("select e.zipFileName from ScannableItem si"
+        + " JOIN Envelope e on e.id = si.envelope.id"
+        + " where si.documentControlNumber like :documentControlNumber"
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ScannableItemRepository.java:20]: with this `like` is testing starting or contains?

**Review by mdayican — COMMENTED** [review]
is there a test for ScannableItemRepository ?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 3]
```diff
@@ -18,15 +24,22 @@
 
 @Service
 public class ZipFileStatusService {
-
+    private static final Logger log = LoggerFactory.getLogger(ReportSender.class);
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:27]: Should be `ZipFileStatusService.class`

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 4]
```diff
@@ -42,6 +55,30 @@ public ZipFileStatus getStatusFor(String zipFileName) {
         );
     }
 
+    public List<ZipFileStatus> getStatusByDcn(String documentControlNumber) throws InvalidParameterException {
+
+        if (documentControlNumber.length() < MIN_LENGTH) {
+            log.error("Exception in Search by DCN error: DCN number specified is less than 6 characters in length.");
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:61]: The other 6s should also be `MIN_LENGTH`

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ScannableItemRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    private EnvelopeRepository envelopeRepo;
+
+    @BeforeEach
+    public void cleanUp() {
+        scannableItemRepo.deleteAll();
+        envelopeRepo.deleteAll();
+    }
+
+    @Test
+    public void should_not_find_complete_envelopes_by_container_if_they_do_not_exist() {
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ScannableItemRepositoryTest.java:43]: why is should not find completed envelopes?

**Review by mdayican — COMMENTED** [review]
can you add test  for ScannableRepo for "contains"

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ScannableItemRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+        // when
+        final List<String> result = scannableItemRepo.findByDcn(documentControlNumber);
+
+        // then
+        assertThat(result).isEqualTo(Arrays.asList("test1.zip", "test2.zip"));
+    }
+
+    @Test
+    public void zip_file_names_should_not_be_found_with_dcn_not_in_beginning() {
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ScannableItemRepositoryTest.java:63]: sth. like `should_not_find_zip_file_when_dcn_not_begins_with_search_string`

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ScannableItemRepositoryTest.java [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    private EnvelopeRepository envelopeRepo;
+
+    @BeforeEach
+    public void cleanUp() {
+        scannableItemRepo.deleteAll();
+        envelopeRepo.deleteAll();
+    }
+
+    @Test
+    public void find_zip_file_names_with_dcn() {
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/ScannableItemRepositoryTest.java:43]: sth. like `should_find_zip_file_when_dcn_begins_with_search_string`

**Review by mdayican — APPROVED** [review]

**Review by apereverzin — APPROVED** [review]

**Pull request merged**